lib/dataset/dataset.cpp

```diff
@@ -146,41 +147,35 @@ void Dataset::setData(const std::string &name, Variable data,
 /// dataset. Throws if there are existing but mismatching coords, masks, or
 /// attributes. Throws if the provided data brings the dataset into an
 /// inconsistent state (mismatching dimensions).
-void Dataset::setData(const std::string &name, const DataArray &data) {
+void Dataset::setData(const std::string &name, DataArray data) {
```

Comment:
No, the previous version pretended to move components of `data` into the dataset. But it didn't actually work because `data` was const. So it triggered copies.
I tried to switch to a copy-and-move idiom but lost a `move` for setting the data.